### DISCUSSION TITLE
Trying to compile source

### DISCUSSION THREAD

**Discussion by IntyLab** [stella-emu/stella#927] [Q&A]
I fetched the source using the "git clone" command, ran the Configure script, and tried to compile the source using "make -f Makefile".  Compilation halted on an error because the file "sys/termios.h" is not found.  This happened in src/os/unix/SerialPortUNIX.cxx at line 22.

I ran the command "find ./src -name termios.h" and got no results.  Is the file just missing from the source package?  Thanks for your help on this.

**Comment by skitt** [comment]
What platform are you building on? `sys/termios.h` is supposed to be provided by the C library.

**Comment by sa666666** [comment]
Yes, sounds like you don't have the C header files installed.  If it's for Ubuntu, perhaps install `build-essentials`?

**Comment by IntyLab** [comment]
Thank you for your assistance.  I found a copy of termios.h by expanding my search outside my home directory, and I modified the two include statements to point to the absolute folder path for that and ioctl.h.

If you'll bear with me while I bring myself up to speed, I'd love to get Stella running in my environment.

In the same file, now I'm getting five "use of undeclared identifier" errors.  I'll keep looking at my end and see if I can figure it out, but I appreciate any help you can offer.

**Reply by skitt** [reply]
Could you tell us what environment you’re trying to get Stella working in? The [build instructions](https://stella-emu.github.io/development.html) should work as-is, you shouldn’t need to change anything in the project.

**Reply by IntyLab** [reply]
The environment is Termux, distributed through F-Droid, not the Google Play Store (because it's no longer supported).  I have SDL2 and a whole X11 desktop environment currently running on it.

What other information do you need?  I appreciate your help with this.

**Comment by IntyLab** [comment]
I'm using Termux, which is available via F-Droid.  I have an X environment, SDL2, and the latest build of clang.

**Comment by IntyLab** [comment]
Here is the output I currently am getting:

mkdir -p -p src/os/unix/.deps
mkdir -p -p out/src/os/unix
g++ -MMD -MF "src/os/unix/.deps/SerialPortUNIX.d" -MQ "out/src/os/unix/SerialPortUNIX.o" -MP -O2 -x c++ -Wall -Wextra -Wno-unused-parameter -Wno-multichar -Wunused -Woverloaded-virtual -Wnon-virtual-dtor -std=c++17 -Weverything -Wno-c++98-compat-pedantic -Wno-unknown-warning-option -Wno-switch-enum -Wno-conversion -Wno-covered-switch-default -Wno-inconsistent-missing-destructor-override -Wno-float-equal -Wno-exit-time-destructors -Wno-global-constructors -Wno-weak-vtables -Wno-four-char-constants -Wno-padded -Wno-reserved-identifier -Wno-duplicate-enum -fomit-frame-pointer -DSDL_SUPPORT -D_GLIBCXX_USE_CXX11_ABI=1 -DUNIX -DBSPF_UNIX -DGUI_SUPPORT -DWINDOWED_SUPPORT -DSOUND_SUPPORT -DDEBUGGER_SUPPORT -DJOYSTICK_SUPPORT -DCHEATCODE_SUPPORT -DHTTP_LIB_SUPPORT -DIMAGE_SUPPORT -DZIP_SUPPORT -Isrc/emucore -Isrc/common -Isrc/common/tv_filters -Isrc/emucore/tia -Isrc/emucore/tia/frame-manager -Isrc/lib/json -Isrc/common/repository/sqlite -I/data/data/com.termux/files/usr/include/SDL2 -I/data/data/com.termux/files/usr/include -DNO_SHARED_MEMORY -D_REENTRANT -D_THREAD_SAFE -Isrc/os/unix -Isrc/gui -Isrc/debugger -Isrc/debugger/gui -Isrc/debugger/yacc -Isrc/cheat -[snip]
src/os/unix/SerialPortUNIX.cxx:47:3: error: use of undeclared identifier 'tcflush'
  tcflush(myHandle, TCOFLUSH);
  ^
src/os/unix/SerialPortUNIX.cxx:48:3: error: use of undeclared identifier 'tcflush'
  tcflush(myHandle, TCIFLUSH);
  ^
src/os/unix/SerialPortUNIX.cxx:49:28: error: use of undeclared identifier 'FNDELAY'
  fcntl(myHandle, F_SETFL, FNDELAY);
                           ^
src/os/unix/SerialPortUNIX.cxx:57:3: error: use of undeclared identifier 'tcflush'
  tcflush(myHandle, TCIFLUSH);
  ^
src/os/unix/SerialPortUNIX.cxx:58:3: error: use of undeclared identifier 'tcsetattr'
  tcsetattr(myHandle, TCSANOW, &termios);
  ^
5 errors generated.
make: *** [Makefile:277: out/src/os/unix/SerialPortUNIX.o] Error 1
